src/web/rustdoc.rs

```diff
@@ -238,15 +238,18 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         return Ok(super::redirect(canonical));
     }
 
-    let path = {
-        let mut path = req_path.join("/");
-        if path.ends_with('/') {
+    if let Some(&end) = req_path.last(){
+        if end == "" {
             req_path.pop(); // get rid of empty string
-            path.push_str("index.html");
             req_path.push("index.html");
+        } else if !end.contains('.') {
```

Comment:
This is not the solution I would like to see. If there is a file (not a directory) called `/:crate/:version/name`, this will prevent it from being seen, instead always redirecting to the directory. I would rather fallback to the directory in https://github.com/rust-lang/docs.rs/blob/777b37cad5ce0812a184f6dcf6603e85ced3cd26/src/web/rustdoc.rs#L254 only if the file does not already exist.

Also, this is a bit of a fragile part of the code and I'd rather not change it unless absolutely necessary.